x/slashing/client/rest/query.go

```diff
@@ -15,10 +15,15 @@ import (
 
 func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *codec.Codec) {
 	r.HandleFunc(
-		"/slashing/validators/{validatorPubKey}/signing_info",
+		"/slashing/validators/signing_info/{validatorPubKey}",
 		signingInfoHandlerFn(cliCtx, slashing.StoreKey, cdc),
 	).Methods("GET")
 
+	r.HandleFunc(
```

Comment:
Will this route include pagination for this data? As a side note this is something we need to add across the SDK.

Reply:
Not yet, but it should!

Regarding this, I was talking with @mossid a couple of weeks ago regarding the feasibility to introduce a middleware strategy that generalise validation of path parameters and pagination across all routes that needs it, to reduce similar code across many routes and to make it easy to remember. Of course it has to be designed beforehand.

Reply:
I was thinking how to achieve that, I pagination available on TransactionSearch only:
```
type SignClient interface {
	Block(height *int64) (*ctypes.ResultBlock, error)
	BlockResults(height *int64) (*ctypes.ResultBlockResults, error)
	Commit(height *int64) (*ctypes.ResultCommit, error)
	Validators(height *int64) (*ctypes.ResultValidators, error)
	Tx(hash []byte, prove bool) (*ctypes.ResultTx, error)
	TxSearch(query string, prove bool, page, perPage int) (*ctypes.ResultTxSearch, error)
}
```

You're right, in REST I think we are doing it in one place where we also keep querying always page 0:
https://github.com/cosmos/cosmos-sdk/blob/424adf14618a65dca1cc47130aa199b7db88688a/x/staking/client/rest/utils.go#L32

Reply:
Should be covered now